src/access-control.interceptor.ts

```diff
@@ -0,0 +1,25 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
+import { Permission } from 'accesscontrol';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+/**
+ * Interceptor that automatically filter attributes defined in access control.
+ */
+@Injectable()
+export class AccessControlInterceptor implements NestInterceptor {
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const { permissions } = context.switchToHttp().getRequest();
```

Comment:
I think I'd prefer to get the metadata from reflection though opposed to setting permissions on the request object as suggested by @cheelahim 

```ts
  protected getPermissions(context: ExecutionContext): Permission[] {
    return this.getHandlerRoles(context).map(role => this.accessControl.permission(role));
  }

  protected getHandlerRoles(context: ExecutionContext): Role[] {
    return this.reflector.get<Role[]>('roles', context.getHandler()) || [];
  }
  ```